synapse/storage/databases/main/events.py

```diff
@@ -491,7 +512,7 @@ def _persist_event_auth_chain_txn(
             table="event_auth_chain_to_calculate",
             keyvalues={},
             column="room_id",
-            iterable={e.room_id for e in state_events.values()},
+            iterable=event_to_room_id.values(),
```

Comment:
Should we be de-duplicating this before throwing it at the database? E.g. what happens if we have the same room ID many times? (Or is that highly unlikely?)

Reply:
Good spot! (Linting said pointless use of generator, but I forgot it didn't mean just remove it entirely...)